### ISSUE TITLE
Push to GHE fails with error: `fatal: early EOF`

### ISSUE THREAD

**Issue by francisschmaltz** [github/codeql-action-sync-tool#43]
Push to GHE kept failing with this error trace:

```
INFO[0000] Starting CodeQL Action sync tool version 1.0.0... 
DEBU[0000] Ensuring repository exists...                
DEBU[0001] Pushing Git releases to https://github.example.com/github/codeql-action.git... 
fatal: early EOF
FATA[0359] stream error: stream ID 13; INTERNAL_ERROR
Error pushing Action to GitHub Enterprise Server.
github.com/github/codeql-action-sync/internal/push.(*pushService).pushGit
	/home/runner/work/codeql-action-sync-tool/codeql-action-sync-tool/internal/push/push.go:193
github.com/github/codeql-action-sync/internal/push.Push
	/home/runner/work/codeql-action-sync-tool/codeql-action-sync-tool/internal/push/push.go:366
github.com/github/codeql-action-sync/cmd.glob..func3
	/home/runner/work/codeql-action-sync-tool/codeql-action-sync-tool/cmd/push.go:16
github.com/spf13/cobra.(*Command).execute
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:842
github.com/spf13/cobra.(*Command).ExecuteC
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950
github.com/spf13/cobra.(*Command).Execute
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
github.com/spf13/cobra.(*Command).ExecuteContext
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:880
github.com/github/codeql-action-sync/cmd.Execute
	/home/runner/work/codeql-action-sync-tool/codeql-action-sync-tool/cmd/root.go:66
main.main
	/home/runner/work/codeql-action-sync-tool/codeql-action-sync-tool/main.go:15
runtime.main
	/opt/hostedtoolcache/go/1.14.7/x64/src/runtime/proc.go:203
runtime.goexit
	/opt/hostedtoolcache/go/1.14.7/x64/src/runtime/asm_amd64.s:1373 
```

The push always seemed to fail after 5 minutes so I tried increasing the timeout time in HAProxy but that had no effect

**Comment by chrisgavin** [comment]
Hi @francisschmaltz.

I've not been able to reproduce this problem but from some research it appears that it can happen when a proxy or firewall sitting between your machine and GitHub Enterprise interferes with communication with the Git server. It's interesting that it fails after 5 minutes as I would not expect pushing the Git releases to take that long.

Can you push commits to GitHub Enterprise over HTTPS using the regular Git command line client? If that also doesn't work I suspect the issue lies somewhere in between the the CodeQL Action Sync Tool and GitHub Enterprise.

I've added a new flag, `--push-ssh` in #45 (now released in v1.0.1) which enables pushing Git contents over SSH instead so that may be an option if you are unable to push via HTTPS.

**Comment by chrisgavin** [comment]
Please let us know if you still have issues with the `--push-ssh` option or more details about any firewalls or proxies running in front of your GitHub Enterprise Server.
